CoreScriptsRoot/ServerCoreScripts/ClientChat/ChannelsBar.lua

```diff
@@ -0,0 +1,222 @@
+		self.ChannelTabs[channelName:lower()] = nil
+
+		self:OrganizeChannelTabs()
+	end
+
+	function obj:SetActiveChannelTab(channelName)
+		if (self.CurrentChannelTab) then
+			self.CurrentChannelTab.ButtonObj.Position = self.BaseChannelTab.ButtonObj.Position
+			self.CurrentChannelTab.ButtonObj.Size = self.BaseChannelTab.ButtonObj.Size
+		end
+		
+		if (channelName) then
+			self.CurrentChannelTab = self.ChannelTabs[channelName:lower()]
+		else
+			self.CurrentChannelTab = nil
+		end
+		
+		if (self.CurrentChannelTab) then
+			self.CurrentChannelTab.ButtonObj.Position = UDim2.new(0, 0, 0, 0)
+			self.CurrentChannelTab.ButtonObj.Size = UDim2.new(1, 0, 1, 0)
+			
+			self.CurrentChannelTab.ButtonObj.TextColor3 = Color3.new(1, 1, 1)
+		end
+
+		self:OrganizeChannelTabs()
+		self.eOnChannelTabChanged:Fire(channelName)
+	end
+
+	function obj:OrganizeChannelTabs()
+		local tabOrder = {}
+		
+		for channelName, tab in pairs(self.ChannelTabs) do
+			local i = 1
+			while (i <= #tabOrder and tabOrder[i] < channelName) do
+				i = i + 1
+			end
+			table.insert(tabOrder, i, channelName)
+		end
+		
+		local posOffsetY = (self.Frame.PlacementFrame.AbsoluteSize.Y - self.Frame.AbsoluteSize.Y)
+		local currentPosition = 1
+		for i, channelName in pairs(tabOrder) do
+			local tab = self.ChannelTabs[channelName]
+			tab.Position = UDim2.new(0, currentPosition, 0, posOffsetY)
+			
+			currentPosition = currentPosition + tab.Size.X.Offset + (tab.BorderSizePixel * 2) + 1
+		end
+		
+		self.Frame.CanvasSize = UDim2.new(0, currentPosition, 0, 0)
+		
+		self:UpdateScrollingBar()
+	end
+
+	function obj:OnMessagePostedInChannel(channelName)
+		local tab = self:GetChannelTab(channelName)
+		if (tab) then
+			if (tab.ButtonObj.Size.Y.Scale ~= 1) then
+				tab.ButtonObj.TextColor3 = self.ChatWindow.ChatSettings.ChannelTabNotificationColor
+			end
+		end
+	end
+	
+	function obj:UpdateTabSizes()
+		local height = self.Frame.AbsoluteSize.Y
+		local minWidth = height * 2
+		
+		for channelName, tab in pairs(self.ChannelTabs) do
+			--tab.Size = UDim2.new(0, 100000, 1, 0)
+			local width = math.max(minWidth, tab.ButtonObj.TextBounds.X + 8)
+			tab.Size = UDim2.new(0, width + (1 * 2), 0, self.Frame.AbsoluteSize.Y)
+			tab.ButtonObj.FontSize = self.FontSize
+		end
+		
+		self:OrganizeChannelTabs()
+	end
+	
+	function obj:UpdateScrollingBar()
+		if (self.Frame.CanvasSize.X.Offset > self.Frame.AbsoluteSize.X) then
+			self.Frame.PlacementFrame.Size = UDim2.new(0, 0, 0.5, 0 - self.Frame.ScrollBarThickness - 2)
+		else
+			self.Frame.PlacementFrame.Size = UDim2.new(0, 0, 0.5, 0)
+		end
+		
+		self.Frame.PlacementFrame.Position = UDim2.new(0, 0, 0.5, 0)
+	end
+	
+	obj.Frame.Changed:connect(function(prop)
+		if (prop == "AbsoluteSize") then
+			obj:UpdateScrollingBar()
+		end
+	end)
+	
+	obj:UpdateScrollingBar()
+	
+	return setmetatable(obj, metatable)
+end
+
+return module
+
+]]
+
+local generated = Instance.new("ModuleScript")
+generated.Name = "Generated"
+generated.Source = source
```

Comment:
These scripts are loaded as instance type of CoreScript, but they need to be ModuleScripts. There isn't a way to load ModuleScripts on the server side in corescripts right now, so I wrote this code to "solve" that problem. It works for now and it's not necessarily all that bad, but it could be changed if this functionality was natively added.